Add proxy peering metrics to docs (#23015)
gravitational/teleport: docs/pages/includes/metrics.mdx

## Proxy Service

| Name                                      | Type      | Component      | Description                                                                                                                     |
| ----------------------------------------- | --------- | -------------- | ------------------------------------------------------------------------------------------------------------------------------- |

> | `failed_connect_to_node_attempts_total`   | counter   | Teleport Proxy | Number of failed SSH connection attempts to a Node. Use with `teleport_connect_to_node_attempts_total` to get the failure rate. |

Reviewer: Does `Node` refer to an SSH Service instance or any agent? We're phasing out the "Node" terminology, and the "Core Concepts" page doesn't mention this term.

| `grpc_client_started_total`               | counter   | Teleport Proxy | Total number of RPCs started on the client.                                                                                     |
| `grpc_client_handled_total`               | counter   | Teleport Proxy | Total number of RPCs completed on the client, regardless of success or failure.                                                 |
| `grpc_client_msg_received_total`          | counter   | Teleport Proxy | Total number of RPC stream messages received on the client.                                                                     |
| `grpc_client_msg_sent_total`              | counter   | Teleport Proxy | Total number of gRPC stream messages sent by the client.                                                                        |
| `proxy_connection_limit_exceeded_total`   | counter   | Teleport Proxy | Number of connections that exceeded the Proxy Service connection limit.                                                         |

> | `proxy_peer_client_dial_error_total`      | counter   | Teleport Proxy | Total number of errors encountered dialing peer proxies.                                                                        |

Reviewer: ```suggestion
| `proxy_peer_client_dial_error_total`      | counter   | Teleport Proxy | Total number of errors encountered dialing peer Proxy Service instances.                                                                        |
```

> | `proxy_peer_server_connections`           | gauge     | Teleport Proxy | Number of currently opened connection to proxy peer servers.                                                                    |

Reviewer: ```suggestion
| `proxy_peer_server_connections`           | gauge     | Teleport Proxy | Number of currently opened connection to peer Proxy Service instances.                                                                    |
```

| `proxy_peer_client_rpc`                   | gauge     | Teleport Proxy | Number of current client RPC requests.                                                                                          |
| `proxy_peer_client_rpc_total`             | counter   | Teleport Proxy | Total number of client RPC requests.                                                                                            |
| `proxy_peer_client_rpc_duration_seconds`  | histogram | Teleport Proxy | Duration in seconds of RPCs sent by the client.                                                                                 |
| `proxy_peer_client_message_sent_size`     | histogram | Teleport Proxy | Size of messages sent by the client.                                                                                            |
| `proxy_peer_client_message_received_size` | histogram | Teleport Proxy | Size of messages received by the client.                                                                                        |

> | `proxy_peer_server_connections`           | gauge     | Teleport Proxy | Number of currently opened connection to proxy peer clients.                                                                    |

Reviewer: ```suggestion
| `proxy_peer_server_connections`           | gauge     | Teleport Proxy | Number of currently opened connection to Proxy Service peer clients.                                                                    |
```

| `proxy_peer_server_rpc_total`             | counter   | Teleport Proxy | Total number of server RPC requests.                                                                                            |
| `proxy_peer_server_rpc_duration_seconds`  | histogram | Teleport Proxy | Duration in seconds of RPCs sent by the server.                                                                                 |
| `proxy_peer_server_message_sent_size`     | histogram | Teleport Proxy | Size of messages sent by the server.                                                                                            |
| `proxy_peer_server_message_received_size` | histogram | Teleport Proxy | Size of messages received by the server.                                                                                        |
| `proxy_ssh_sessions_total`                | gauge     | Teleport Proxy | Number of active sessions through this Proxy Service instance.                                                                  |

> | `proxy_missing_ssh_tunnels`               | gauge     | Teleport Proxy | Number of missing SSH tunnels. Used to debug if Nodes have discovered all Proxy Service instances.                              |

Reviewer: Same comment as above re: "Node"

| `remote_clusters`                         | gauge     | Teleport Proxy | Number of inbound connections from leaf clusters.                                                                               |

> | `teleport_connect_to_node_attempts_total` | counter   | Teleport Proxy | Number of SSH connection attempts to a node. Use with `failed_connect_to_node_attempts_total` to get the failure rate.          |

Reviewer: Same comment as above re: "node"

| `proxy_peer_client_rpc`                   | gauge     | Teleport Proxy | Number of current client RPC requests.                                                                                                   |
| `proxy_peer_client_rpc_total`             | counter   | Teleport Proxy | Total number of client RPC requests.                                                                                                     |
| `proxy_peer_client_rpc_duration_seconds`  | histogram | Teleport Proxy | Duration in seconds of RPCs sent by the client.                                                                                          |
| `proxy_peer_client_message_sent_size`     | histogram | Teleport Proxy | Size of messages sent by the client.                                                                                                     |
| `proxy_peer_client_message_received_size` | histogram | Teleport Proxy | Size of messages received by the client.                                                                                                 |

> | `proxy_peer_server_connections`           | gauge     | Teleport Proxy | Number of currently opened connection to proxy peer clients.                                                                             |

Reviewer: ```suggestion
| `proxy_peer_server_connections`           | gauge     | Teleport Proxy | Number of currently opened connection to peer Proxy Service clients.                                                                             |
```
